src/main/java/net/minestom/server/entity/Player.java

```diff
@@ -248,13 +250,38 @@ public Player(@NotNull UUID uuid, @NotNull String username, @NotNull PlayerConne
     public CompletableFuture<Void> UNSAFE_init(@NotNull Instance spawnInstance) {
         this.dimensionType = spawnInstance.getDimensionType();
 
-        NBTCompound nbt = NBT.Compound(Map.of(
-                "minecraft:chat_type", Messenger.chatRegistry(),
-                "minecraft:dimension_type", MinecraftServer.getDimensionTypeManager().toNBT(),
-                "minecraft:worldgen/biome", MinecraftServer.getBiomeManager().toNBT()));
+        var registry = new HashMap<String, NBT>();
```

Comment:
I'd suggest keeping the previous code initiating the immutable compound, but with the `minecraft:damage_type` created just above. As the conversions are a bit unnecessary.